main.go

```diff
@@ -0,0 +1,274 @@
+	"crud-golang/internal/dto"
+	"crud-golang/internal/repositories/mongo"
+	services "crud-golang/internal/services"
+	"crud-golang/internal/services/database"
+	"crud-golang/pkg/client/mongodb"
+	"crud-golang/pkg/logging"
+	"crud-golang/pkg/responses"
+	swagger "github.com/arsmn/fiber-swagger"
+	"github.com/go-playground/validator/v10"
+	"github.com/gofiber/fiber"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+)
+
+var (
+	repository = mongo.NewRepository()
+	service    = database.NewService(repository)
+	validate   = validator.New()
+)
+
+// @title CRUD account
+// @version 1.0
+// @description API Server for CRUD Application
+
+// @host localhost:8181
+// @BasePath /accounts
+
+// @securityDefinitions.apikey ApiKeyAuth
+// @in header
+// @name Authorization
+func main() {
+	logging.NewLogger()
+	_, err := mongodb.ConnectDB()
+	if err != nil {
+		return
+	}
+	app := fiber.New()
+
+	app.Post("/login", signIn)
+	accountApi := app.Group("/accounts")
+	accountApi.Get("/", findAll)
+	accountApi.Post("/", createAccount)
+	accountApi.Get("/:accountId", findOne)
+	accountApi.Put("/:accountId", updateAccount)
+	accountApi.Delete("/:accountId", deleteAccount)
+
+	//app.Use("/hello", func(ctx *fiber.Ctx) {
+	//	jwtware.New(jwtware.Config{
+	//		SigningKey: []byte(os.Getenv("SECRET")),
+	//	})
+	//
+	//	ctx.Send("Hello, world!")
+	//})
+	app.Get("/private", private)
+
+	registerSwagger(app)
+
+	app.Listen(":8181")
+}
+
+func private(ctx *fiber.Ctx) {
+	authorization := ctx.Fasthttp.Request.Header.Peek("Authorization")
+	_, str, _ := strings.Cut(string(authorization), "Bearer ")
+
+	prime, err := services.VerifyJWT(str)
+	if err != nil {
+		ctx.Status(fiber.StatusUnauthorized)
+	}
+	if prime {
+		ctx.JSON(fiber.Map{"Data": "Hello world!"})
+	}
+}
+
+func registerSwagger(app *fiber.App) {
+	enableSwagger := os.Getenv("ENABLE_SWAGGER")
+	if enabled, _ := strconv.ParseBool(enableSwagger); enabled {
+		route := app.Group("/swagger")
+		route.Get("*", swagger.Handler)
+		logging.Infof("Swagger Started")
+	}
+}
+
+// @Summary SignIn
+// @Tags auth
+// @Description create account
+// @ID signin
+// @Accept  json
+// @Produce  json
+// @Param input body dto.Login true "account info"
+// @Success 200 {integer} integer 1
+// @Failure 400,404 {object} HTTPError
+// @Failure 500 {object} HTTPError
+// @Failure default {object} HTTPError
+// @Router /auth/sign-up [post]
+func signIn(ctx *fiber.Ctx) {
+	var accountDto dto.Login
+
+	if err := ctx.BodyParser(&accountDto); err != nil {
+		ctx.Status(http.StatusBadRequest).JSON(responses.Account{Status: http.StatusBadRequest,
+			Message: "error", Data: &fiber.Map{"data": err.Error()}})
+		return
+	}
+
+	if validationErr := validate.Struct(&accountDto); validationErr != nil {
+		ctx.Status(http.StatusBadRequest).JSON(responses.Account{Status: http.StatusBadRequest,
+			Message: "error", Data: &fiber.Map{"data": validationErr.Error()}})
+		return
+	}
+
+	account, err := service.Login(accountDto.Email, accountDto.Password)
```

Comment:
You should leverage request context here. If the request is being cancelled, ideally, you would like to cancel operations since there will be no one to receive it. Each service method should accept ctx as the first argument and either make child contexts, or propagate it so you have proper cancelation that is request based